test/itemrenderer.cpp

```diff
@@ -0,0 +1,190 @@
+#include "3rd-party/catch.hpp"
+
+#include "cache.h"
+#include "configcontainer.h"
+#include "itemrenderer.h"
+#include "test-helpers.h"
+
+using namespace newsboat;
+
+static const auto ITEM_TITLE = std::string("A frivolous test item");
+static const auto ITEM_AUTHOR = std::string("Johnny Doe Jr.");
+// Sun Sep 30 19:34:25 UTC 2018
+static const auto ITEM_PUBDATE = time_t{1538336065};
+static const auto ITEM_LINK = std::string("https://example.com/see-more");
+static const auto ITEM_DESCRIPTON = std::string("<p>Hello, world!</p>");
+static const auto ITEM_DESCRIPTON_RENDERED = std::string("Hello, world!");
+static const auto ITEM_ENCLOSURE_URL =
+	std::string("https://example.com/see-more.mp3");
+std::shared_ptr<RssItem> test_item(Cache* c)
+{
+	auto item = std::make_shared<RssItem>(c);
+
+	item->set_title(ITEM_TITLE);
+	item->set_author(ITEM_AUTHOR);
+	item->set_pubDate(ITEM_PUBDATE);
+	item->set_link(ITEM_LINK);
+
+	return item;
+}
+
+TEST_CASE("to_plain_text() produces a rendered representation of RSS item, "
+		"ready to be displayed to the user",
+		"[ItemRenderer]")
+{
+	TestHelpers::EnvVar tzEnv("TZ");
+	tzEnv.set("UTC");
+
+	ConfigContainer cfg;
+	// ItemRenderer uses that setting, so let's fix its value to make the test
+	// reproducible
+	cfg.set_configvalue("text-width", "80");
+
+	Cache rsscache(":memory:", &cfg);
+	ItemRenderer renderer(&cfg);
+
+	auto item = test_item(&rsscache);
+
+	SECTION("Item without an enclosure") {
+		item->set_description(ITEM_DESCRIPTON);
+
+		const auto result = renderer.to_plain_text(item);
+
+		const auto expected = std::string() +
+			"Title: " + ITEM_TITLE + '\n' +
```

Comment:
Disclaimer: I didn't try it, so I could be totally wrong on that one. As part of the message is localized (`Title: `), should we not also explicitly set the `LANG` environment variable or the like to make sure the English translations will be used? Otherwise tests may fail on non-EN systems.

Reply:
"test/test.cpp" doesn't set text domains like "newsboat.cpp" does, so gettext is effectively disabled in tests, meaning things like `_("Title")` are guaranteed to be turned into `"Title"`.

But I didn't verify that because I don't have Newsboat installed at the moment. One needs it installed to check this, otherwise /usr/share/locale/ won't have Newsboat messages in it, and all gettext calls will return default strings (e.g. "Title").

Reply:
That's good to know.